File: reactive_mongodb_example_java/src/main/java/com/github/nkolytschew/data/repository/custom/ConfigurationDocumentRepositoryCustom.java
package com.github.nkolytschew.data.repository.custom;

import com.github.nkolytschew.data.document.helper.ConfigurationWithRatings;

import org.springframework.stereotype.Repository;

import java.util.List;

public interface ConfigurationDocumentRepositoryCustom {

	List<ConfigurationWithRatings> getAggregatedConfigurationWithRatingsById(String configurationId);

}

File: reactive_mongodb_example/src/main/kotlin/com/github/nkolytschew/data/repository/custom/ConfigurationDocumentRepositoryCustom.kt
package com.github.nkolytschew.data.repository.custom

import com.github.nkolytschew.data.document.helper.ConfigurationWithRatings
import reactor.core.publisher.Flux

interface ConfigurationDocumentRepositoryCustom {

  fun getAggregatedConfigurationWithRatingsById(configurationId: String): List<ConfigurationWithRatings>

}